src/cpu.rs

```diff
@@ -17,23 +20,222 @@ pub struct CPU {
     pub r7: u16,
     pub pc: u16,
     pub cond: u16,
+    pub memory: Memory,
 }
 
 impl CPU {
     pub fn new() -> Self {
         Self {
+            r0: 0,
+            r1: 0,
+            r2: 0,
+            r3: 0,
+            r4: 0,
+            r5: 0,
+            r6: 0,
+            r7: 0,
             pc: 0x3000,
-            ..Default::default()
+            cond: 0,
+            memory: Memory::new(),
         }
     }
 
-    pub fn fetch_instruction(&mut self, memory: &Memory) -> Option<u16> {
-        let instruction = memory.read(self.pc.into())?;
+    pub fn fetch_instruction(&mut self) -> Option<u16> {
+        let instruction = self.memory.read(self.pc.into())?;
         self.pc = self.pc.checked_add(1)?;
 
         let op = Opcode::from(instruction);
-        println!("{:#?}", op);
 
         Some(instruction)
     }
+
+    pub fn execute(&mut self, opcode: Opcode) -> Result<(), CPUErrors> {
+        match opcode {
+            Opcode::OP_ADD_SR { dr, sr1, sr2 } => {
```

Comment:
What do you think about naming this opcode `OP_ADD_REG` instead? I believe it signifies better what the instructions does.

Very nit-picky, and it's totally subjective though.